src/OAuth2/Encryption/FirebaseJwt.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace OAuth2\Encryption;
+
+/**
+ * Bridge file to use the firebase/php-jwt package for JWT encoding and decoding.
+ * @author Francis Chuang <francis.chuang@gmail.com>
+ */
+class FirebaseJwt implements EncryptionInterface
+{
+    public function __construct()
+    {
+        if (!class_exists('\JWT')) {
+            throw new \ErrorException('firebase/php-jwt must be installed to use this feature. You can do this by running "composer require firebase/php-jwt"');
+        }
+    }
+
+    public function encode($payload, $key, $alg = 'HS256', $keyId = null)
+    {
+        return \JWT::encode($payload, $key, $alg, $keyId);
+    }
+
+    public function decode($jwt, $key = null, $allowedAlgorithms = null)
+    {
+        try {
+
+            //Maintain BC: Do not verify if no algorithms are passed in.

```

Comment:
In `firebase/php-jwt` `allowedAlgorithms` should not be anything but an array (and it doesn't make sense for it to be anything but an array). In this case, I believe that we should update the interface to force `allowedAlgorithms` to be an array: `array $allowedAlogrithms = array()`. This allows us to remove the code to maintain BC and also provides type hinting in IDEs.